src/system/SystemPacketBuffer.cpp

```diff
@@ -707,5 +712,19 @@ PacketBuffer * PacketBuffer::BuildFreeList()
 
 #endif //  !CHIP_SYSTEM_CONFIG_USE_LWIP && CHIP_SYSTEM_CONFIG_PACKETBUFFER_MAXALLOC
 
+PacketBufferHandle PacketBufferHandle::DetachTail()
```

Comment:
So this has quite different semantics from `PacketBuffer::DetachTail`: it returns the old head, not the old tail, and updates the identity of `this` rather than keeping that identity.

It seems to me that this basically does what `FreeHead` does, but instead of freeing the head it returns it.  Maybe it should be called `TakeHead` or `PopHead` or some such?  Calling it `DetachTail` really doesn't make sense, since the thing it's being called on becomes what used to be the tail.

Reply:
Changed to PopHead().